server/parsing_lib/log_parser.py

```diff
@@ -8,68 +8,135 @@
 
 
 class LogParser(object):
-    """ Handles parsing of all logs from the Mobile API. """
+    """ Handles parsing of all logs from the Mobile API.
+
+    Attributes:
+        parser_info: info required by the parser that is specific to each OS
+    """
+    parser_info = {
+        'Android': {
+            'logLineRegex': re.compile(
+                '(.*)\\s+(\\d*)\\s+(\\d*) ([IWVEDAF]) (.*?): ((?:.*\\n*)*)'),
+            'datetimeFormat': ['%Y-%m-%d %H:%M:%S.%f'],
+            'filterLineRegex': re.compile('-* beginning of'),
+            'groupNums': {
+                'time': 1,
+                'processId': 2,
+                'threadId': 3,
+                'logType': 4,
+                'tag': 5,
+                'text': 6,
+            },
+            'logTypes': {
+                'I': 'Info',
+                'W': 'Warning',
+                'V': 'Verbose',
+                'E': 'Error',
+                'D': 'Debug',
+                'A': 'WTF',
+                'F': 'Fatal',
+            },
+        },
+        'iOS': {
+            'logLineRegex': re.compile('(.*) (.*)\\[(\\d+):(\\d+)\\] (.*)'),
+            'datetimeFormat': [
+                '%Y-%m-%d %H:%M:%S.%f',
+                '%Y-%m-%d %H:%M:%S.%f%z',
+                '%Y-%m-%d %H:%M:%S %z'
+            ],
+            'exceptionRegex': re.compile('(.*?)-{2,} BEGIN UNHANDLED EXCEPTION'),
+            'groupNums': {
```

Comment:
Would it make sense to match the iOS numbers with the Android ones?
time 1
processId 2
threadId 3
etc.